src/runtime/converter.cs

```diff
@@ -893,12 +894,89 @@ private static bool ToArray(IntPtr value, Type obType, out object result, bool s
                 Runtime.XDecref(item);
             }
             Runtime.XDecref(IterObject);
+            result = list;
+            return true;
+        }
+
+        /// <summary>
+        /// Convert a Python value to a correctly typed managed array instance.
+        /// The Python value must support the Python iterator protocol or and the
+        /// items in the sequence must be convertible to the target array type.
+        /// </summary>
+        private static bool ToArray(IntPtr value, Type obType, out object result, bool setError)
+        {
+            Type elementType = obType.GetElementType();
+            result = null;
+            bool success = ToListOfT(value, elementType, out result);
+            if (!success)
+            {
+                if (setError)
+                    SetConversionError(value, obType);
+                return false;
+            }
 
-            items = Array.CreateInstance(elementType, list.Count);
+            IList list = (IList)result;
+            Array items = Array.CreateInstance(elementType, list.Count);
             list.CopyTo(items, 0);
-            
+
             result = items;
             return true;
+            
+        }
+
+        /// <summary>
+        /// Convert a Python function to a System.Action.
+        /// The Python value must support the Python "BLAH" protocol.
+        /// </summary>
+        private static bool ToAction(IntPtr value, Type obType, out object result, bool setError) {
+
+            result = null;
+
+            PyObject obj = new PyObject(value);
+
+            var args = obType.GetGenericArguments();
+
+            //Temporarily only deal with non-generic actions!
+            if (!obj.IsCallable() || args.Length != 0) {
+                if (setError) {
+                    SetConversionError(value, obType);
+                }
+                obj.Dispose();
+                return false;
+            }
+
+            Action action = () => {
+                PyObject py_action = new PyObject(value);
+                var py_args = new PyObject[0];
+                var py_result = py_action.Invoke(py_args);
+
+                //Discard the result since this is being converted to an Action
+                py_result.Dispose();
+                py_action.Dispose();
+            };
```

Comment:
The object pointed to by `value` can be collected between the construction of this `action`, and the moment it is invoked. It would lead to a memory corruption.

Reply:
I had forgotten to push the latest change set, please check again